src/reflect/value.go

```diff
@@ -794,7 +890,30 @@ func (v Value) Convert(t Type) Value {
 }
 
 func MakeSlice(typ Type, len, cap int) Value {
-	panic("unimplemented: reflect.MakeSlice()")
+	if typ.Kind() != Slice {
+		panic("reflect.MakeSlice of non-slice type")
+	}
+	if len < 0 {
+		panic("reflect.MakeSlice: negative len")
+	}
+	if cap < 0 {
+		panic("reflect.MakeSlice: negative cap")
+	}
+	if len > cap {
+		panic("reflect.MakeSlice: len > cap")
+	}
+
+	var slice sliceHeader
+	slice.cap = uintptr(cap)
+	slice.len = uintptr(len)
```

Comment:
To be extra safe it might be a good idea to check whether these values actually fit:

```go
if int(uintptr(cap)) != cap { panic() }
```

This is only really relevant on AVR, and should be a no-op on other architectures (it should be trivial for the compiler to optimize this check away on 32-bit and 64-bit systems).